### PR TITLE
App Configuration Sync Token Push Refresh

### PR DESCRIPTION
**Pull request by mrm9084**
# Description
* Added Sync Token to Push Refresh
* Removes domain from push refresh check

### CODE REVIEW

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java [hunk 1]
```diff
@@ -81,15 +85,24 @@ public String refresh(HttpServletRequest request, HttpServletResponse response,
...
         JsonNode validationResponse = kvReference.findValue(VALIDATION_CODE_KEY);
         if (validationResponse != null) {
             // Validating Web Hook
             return String.format("%s%s\"}", VALIDATION_CODE_FORMAT_START, validationResponse.asText());
         } else {
             if (contextRefresher != null) {
                 if (validation.triggerRefresh()) {
-                    publisher.publishEvent(
-                        new AppConfigurationRefreshEvent(validation.getEndpoint()));
+                    publisher.publishEvent(new AppConfigurationRefreshEvent(validation.getEndpoint(), syncToken));
```
Comment (saragluna) [inline sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java:105]: is it possible that the `syncToken` is null?

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/AppConfigurationEndpoint.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public AppConfigurationEndpoint(JsonNode request, List<ConfigStore> configStores
         if (requestTopic != null) {
             String topic = requestTopic.asText();
             store = topic.substring(topic.toLowerCase(Locale.ROOT).indexOf(CONFIG_STORE_TOPIC) + CONFIG_STORE_TOPIC.length() + 1);
-            endpoint = String.format("https://%s.azconfig.io", store);
```
Comment (saragluna) [inline sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/AppConfigurationEndpoint.java:48]: why do we make such change?

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java [hunk 3]
```diff
@@ -81,15 +85,24 @@ public String refresh(HttpServletRequest request, HttpServletResponse response,
...
+        String syncToken = "";
+
+        JsonNode data = kvReference.findValue(DATA);
+        if (data != null) {
+            JsonNode syncTokenNode = data.findValue(SYNC_TOKEN);
+            if (syncTokenNode != null) {
+                syncToken = syncTokenNode.asText();
+            }
+        }
+
```
Comment (saragluna) [inline sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java:97]: This seems like duplicate code with the one in `AppConfigurationBusRefreshEndpoint`, maybe we could create a util class and extract it into a method?

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java (hunk 1 above)
Comment (mrm9084) [reply sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java]: It shouldn't but I added a null check before it is used.

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java (hunk 3 above)
Comment (mrm9084) [reply sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java]: I reworked this.

**Review by saragluna — APPROVED** [review]

**Pull request merged**